src/dreamchecker/lib.rs

```diff
@@ -1281,6 +1281,11 @@ impl<'o, 's> AnalyzeProc<'o, 's> {
             },
             Statement::Return(Some(expr)) => {
                 // TODO: factor in the previous return type if there was one
+                if self.inside_newcontext > 0 {
```

Comment:
IIRC isn't this also set in waitfor = 0 procs?

Reply:
I don't think so, the only place that changes `inside_newcontext` to non-zero values seems to be the `Statement::Spawn` match arm.